near2eth/near2eth-block-relay/index.js

```diff
@@ -14,13 +15,251 @@ const {
+  'eth-ed25519-bin-path',
+  'Path to the .bin file defining Ethereum ED25519 contract.',
+  path.join(LIBS_SOL_SRC_DIR, 'nearbridge/dist/Ed25519.full.bin')
+)
+RainbowConfig.declareOption(
+  'eth-client-lock-eth-amount',
+  'Amount of Ether that should be temporarily locked when submitting a new header to EthClient, in wei.',
+  '100000000000000000000'
+)
+RainbowConfig.declareOption(
+  'eth-client-lock-duration',
+  'The challenge window during which anyone can challenge an incorrect ED25519 signature of the Near block, in EthClient, in seconds.',
+  14400
+)
+RainbowConfig.declareOption(
+  'eth-client-replace-duration',
+  'Minimum time difference required to replace a block during challenge period, in EthClient, in seconds.',
+  18000
+)
+RainbowConfig.declareOption(
+  'eth-client-address',
+  'ETH address of the EthClient contract.'
+)
+RainbowConfig.declareOption(
+  'eth-client-abi-path',
+  'Path to the .abi file defining Ethereum Client contract.',
+  path.join(LIBS_SOL_SRC_DIR, 'nearbridge/dist/NearBridge.full.abi')
+)
+RainbowConfig.declareOption(
+  'eth-client-bin-path',
+  'Path to the .bin file defining Ethereum Client contract.',
+  path.join(LIBS_SOL_SRC_DIR, 'nearbridge/dist/NearBridge.full.bin')
+)
+RainbowConfig.declareOption(
+  'eth-prover-address',
+  'ETH address of the EthProver contract.'
+)
+RainbowConfig.declareOption(
+  'eth-prover-abi-path',
+  'Path to the .abi file defining Ethereum Prover contract.',
+  path.join(LIBS_SOL_SRC_DIR, 'nearprover/dist/NearProver.full.abi')
+)
+RainbowConfig.declareOption(
+  'eth-prover-bin-path',
+  'Path to the .bin file defining Ethereum Prover contract.',
+  path.join(LIBS_SOL_SRC_DIR, 'nearprover/dist/NearProver.full.bin')
+)
+RainbowConfig.declareOption(
+  'near2eth-relay-min-delay',
+  "Minimum number of seconds to wait if the relay can't submit a block right away.",
+  '1'
+)
+RainbowConfig.declareOption(
+  'near2eth-relay-max-delay',
+  "Maximum number of seconds to wait if the relay can't submit a block right away.",
+  '600'
+)
+RainbowConfig.declareOption(
+  'near2eth-relay-error-delay',
+  'Number of seconds to wait before retrying if there is an error.',
+  '1'
+)
+RainbowConfig.declareOption(
+  'watchdog-delay',
+  'Number of seconds to wait after validating all signatures.',
+  '300'
+)
+RainbowConfig.declareOption(
+  'watchdog-error-delay',
+  'Number of seconds to wait before retrying if there is an error.',
+  '1'
+)
+RainbowConfig.declareOption('near-erc20-account', 'Must be declared before set')
+
 class Near2EthRelay {
   async initialize () {
     // @ts-ignore
     this.robustWeb3 = new RobustWeb3(RainbowConfig.getParam('eth-node-url'))
     this.web3 = this.robustWeb3.web3
     this.ethMasterAccount = this.web3.eth.accounts.privateKeyToAccount(
-      normalizeEthKey(RainbowConfig.getParam('eth-master-sk'))
+      // TODO remove this hard-coded constant!
+      normalizeEthKey('0x2bdd21761a483f71054e14f5b827213567971c676928d9a1808cbfa4b7501201')
```

Comment:
Why do we need hardcoded constant instead of receiving it as argument?

Reply:
Yep, this is bad. We should use config there when it will be available. For now I will pass it as argument.

Reply:
Ok, I moved it into `runNear2EthRelay` temporary. This should be fine for now. It will be changed when new config will come.